app/src/main/java/org/schabi/newpipe/fragments/list/search/filter/SearchFilterDialogSpinnerAdapter.java

```diff
@@ -0,0 +1,220 @@
+// Created by evermind-zz 2022, licensed GNU GPL version 3 or later
+
+package org.schabi.newpipe.fragments.list.search.filter;
+
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.view.Gravity;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.BaseAdapter;
+import android.widget.Spinner;
+import android.widget.TextView;
+
+import org.schabi.newpipe.extractor.search.filter.FilterContainer;
+import org.schabi.newpipe.extractor.search.filter.FilterGroup;
+import org.schabi.newpipe.extractor.search.filter.FilterItem;
+import org.schabi.newpipe.util.ServiceHelper;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.schabi.newpipe.fragments.list.search.filter.SearchFilterOptionMenuAlikeDialogGenerator.dipToPixels;
+
+public class SearchFilterDialogSpinnerAdapter extends BaseAdapter {
+
+    private final Context context;
+    private final FilterGroup group;
+    private final BaseSearchFilterUiGenerator.UiWrapperMapDelegate wrapperDelegate;
+    private final Spinner spinner;
+    private final Map<Integer, Integer> id2PosMap = new HashMap<>();
+    private final Map<Integer, UiItemWrapperSpinner>
+            viewWrapperMap = new HashMap<>();
```

Comment:
You might want to use [`SparseIntArray`](https://developer.android.com/reference/android/util/SparseIntArray) and [`SparseArray`](https://developer.android.com/reference/android/util/SparseArray) / [`SparseArrayCompat`](https://developer.android.com/reference/androidx/collection/SparseArrayCompat) instead to reduce memory usage.

Reply:
I've converted all of them to SparseArrayCompat even though I could use SparseIntArray to some extend.
The problem is that my unit test otherwise has to be converted to androidTest and that would slow testing down so I wanted to avoid it. Another options would be to use roboelectric. If adding roboelectric to the test dependencies would be an option let me know.